addons/service/system/dispmanx_vnc/package.mk

```diff
@@ -0,0 +1,86 @@
+#  the Free Software Foundation; either version 2, or (at your option)
+#  any later version.
+#
+#  This Program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with OpenELEC.tv; see the file COPYING.  If not, write to
+#  the Free Software Foundation, 51 Franklin Street, Suite 500, Boston, MA 02110, USA.
+#  http://www.gnu.org/copyleft/gpl.html
+################################################################################
+
+PKG_NAME="dispmanx_vnc"
+PKG_ADDON_NAME="Raspberry Pi VNC"
+PKG_VERSION="a061e0a"
+PKG_REV="0"
+PKG_ARCH="arm"
+PKG_LICENSE="OSS"
+PKG_SITE="https://github.com/hanzelpeter/dispmanx_vnc"
+PKG_URL="$DISTRO_SRC/${PKG_NAME}-${PKG_VERSION}.tar.xz"
+PKG_DEPENDS_TARGET="toolchain libvncserver bcm2835-driver"
+PKG_PRIORITY="optional"
+PKG_SECTION="service/system"
+PKG_SHORTDESC="VNC Server for Raspberry PI using dispmanx"
+PKG_LONGDESC="VNC Server for Raspberry PI using dispmanx"
+PKG_DISCLAIMER="this is an unofficial addon. please don't ask for support in openelec forum / irc channel"
+
+PKG_IS_ADDON="yes"
+PKG_ADDON_TYPE="xbmc.service"
+PKG_ADDON_PROVIDES=""
+
+PKG_AUTORECONF="no"
+
+PKG_MAINTAINER="Lukas Rusak (lrusak at irc.freenode.net)"
+
+pre_build_target() {
+  mkdir -p $ROOT/$PKG_BUILD/.$TARGET_NAME
+    cp -RP $ROOT/$PKG_BUILD/* $ROOT/$PKG_BUILD/.$TARGET_NAME
+}
+
+make_target() {  
+  cd $ROOT/$PKG_BUILD/.$TARGET_NAME
+  
+  $CC $CFLAGS main.c -o dispmanx_vncserver -DHAVE_LIBBCM_HOST \
+                                           -DUSE_EXTERNAL_LIBBCM_HOST \
+                                           -DUSE_VCHIQ_ARM \
+                                           $LDFLAGS \
+                                           -I$SYSROOT_PREFIX/include \
+                                           -I$SYSROOT_PREFIX/usr/include \
+                                           -I$SYSROOT_PREFIX/usr/include/interface/vcos/pthreads \
+                                           -I$SYSROOT_PREFIX/usr/include/interface/vmcs_host \
+                                           -I$SYSROOT_PREFIX/usr/include/interface/vmcs_host/linux \
+                                           -L$SYSROOT_PREFIX/usr/lib\
+                                           -L$SYSROOT_PREFIX/lib \
+                                           -lGLESv2 \
+                                           -lEGL \
+                                           -lopenmaxil \
+                                           -lbcm_host \
+                                           -lvcos \
+                                           -lvchiq_arm \
+                                           -lpthread \
+                                           -lrt \
+                                           -lvncserver
```

Comment:
hint for static build:
```
....
                                           -lrt \
                                           -lz \
                                           -lssl -lcrypto \
                                           -lresolv \
                                           -lvncserver \
                                           -ljpeg -lpng16
```
the order is important.

readelf-d should report:
```
 0x00000001 (NEEDED)                     Shared library: [libbcm_host.so]
 0x00000001 (NEEDED)                     Shared library: [libpthread.so.0]
 0x00000001 (NEEDED)                     Shared library: [libz.so.1]
 0x00000001 (NEEDED)                     Shared library: [libssl.so.27]
 0x00000001 (NEEDED)                     Shared library: [libcrypto.so.30]
 0x00000001 (NEEDED)                     Shared library: [libresolv.so.2]
 0x00000001 (NEEDED)                     Shared library: [libgcc_s.so.1]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
```

make sure to remove libvnc*.so* from your toolchain when switching from shared to static.